core/auth/src/main/java/software/amazon/awssdk/auth/credentials/AwsSessionCredentials.java

```diff
@@ -96,6 +110,38 @@ public int hashCode() {
         hashCode = 31 * hashCode + Objects.hashCode(accessKeyId());
         hashCode = 31 * hashCode + Objects.hashCode(secretAccessKey());
         hashCode = 31 * hashCode + Objects.hashCode(sessionToken());
+        hashCode = 31 * hashCode + Objects.hashCode(accountId);
         return hashCode;
     }
+
+    public static final class Builder {
```

Comment:
Should there be any relation between the session credential's builder and the non-session credential's builder?

Reply:
The two classes are unrelated and old, so I didn't put in the effort. Do you think it's worth being more thorough, and  if so why?

Reply:
The two classes are related, since AwsSessionCredentials extends AwsCredentials, and the AwsBasicCredentials's builder is essentially the builder for AwsCredentials. It's just in a separate class for *reasons*.

Something being old isn't a great reason not to put in the effort. I guess since this will be deprecated eventually then that's a better reason. In the new types it would be nice to have them share an interface in case someone has generic logic that configures both credential types.